akka-actor-tests/src/test/scala/akka/util/WildcardTreeSpec.scala

```diff
@@ -0,0 +1,85 @@
+      tree.find(List("a", "b")) shouldBe leafTree(2)
+      tree.find(List("a")) shouldBe emptyTree.insert(Nil, 1).insert("b", 2).insert("c", 3)
+      tree.find(List("x")) shouldBe emptyTree
+      tree.find(Nil) shouldBe tree
+    }
+
+    "match all elements in the subpath when it contains a wildcard" in {
+      val tree1 = emptyTree.insert("a", 1).insert(List("a", "*"), 1)
+      tree1.find(List("z")) shouldBe emptyTree
+      tree1.find(List("a")) shouldBe leafTree(1).insert("*", 1)
+      tree1.find(List("a", "b")) shouldBe leafTree(1)
+      tree1.find(List("a", "x")) shouldBe leafTree(1)
+
+      val tree2 = emptyTree.insert(List("a", "*", "c"), 1)
+      tree2.find(List("z")) shouldBe emptyTree
+      tree2.find(List("a", "b")) shouldBe emptyTree.insert("c", 1)
+      tree2.find(List("a", "x")) shouldBe emptyTree.insert("c", 1)
+      tree2.find(List("a", "x", "c")) shouldBe leafTree(1)
+      tree2.find(List("a", "x", "y")) shouldBe emptyTree
+    }
+
+    "never find anything when empty" in {
+      emptyTree.find("a") shouldBe emptyTree
+      emptyTree.find(List("a", "b")) shouldBe emptyTree
+      emptyTree.find(Nil) shouldBe emptyTree
+    }
+
+    "recursively match all elements in the subpaths when it contains a double wildcard" in {
+      val tree1 = emptyTree.insert(List("a", "**"), 1)
+      tree1.find(List("z")) shouldBe emptyTree
+      tree1.find(List("a", "b")) shouldBe leafTree(1)
+      tree1.find(List("a", "x")) shouldBe leafTree(1)
+      tree1.find(List("a", "x", "y")) shouldBe leafTree(1)
+
+      val tree2 = emptyTree.insert(List("a", "**", "c"), 1)
+      tree2.find(List("z")) shouldBe emptyTree
+      tree2.find(List("a", "b")) shouldBe emptyTree.insert("c", 1)
+      tree2.find(List("a", "x")) shouldBe emptyTree.insert("c", 1)
+      tree2.find(List("a", "x", "y", "c")) shouldBe leafTree(1)
+      tree2.find(List("a", "x", "y")) shouldBe emptyTree.insert("c", 1)
+
+      val tree3 = emptyTree.insert(List("**", "b"), 1)
+      tree3.find("a") shouldBe emptyTree.insert("b", 1)
+      tree3.find("b") shouldBe emptyTree.insert("b", 1)
+      tree3.find(List("a", "b")) shouldBe leafTree(1)
+      tree3.find(List("a", "x", "b")) shouldBe leafTree(1)
+      tree3.find(List("a", "b", "c")) shouldBe emptyTree.insert("b", 1)
+
+      val tree4 = emptyTree.insert(List("**", "b", "**"), 1)
+      tree4.find("a") shouldBe emptyTree.insert(List("b", "**"), 1)
+      tree4.find(List("a", "b")) shouldBe emptyTree.insert("**", 1)
+      tree4.find(List("a", "x", "b")) shouldBe emptyTree.insert("**", 1)
+      tree4.find(List("a", "b", "c")) shouldBe leafTree(1)
+
+      val tree5 = emptyTree.insert(List("**"), 1)
+      tree5.find(List("anything", "I", "want")) shouldBe leafTree(1)
+      tree5.find("anything") shouldBe leafTree(1)
+    }
+  }
+
+  private implicit def strToIterator(s: String): Iterator[String] = List(s).iterator
+  private implicit def listToIterator[T](l: List[T]): Iterator[T] = l.iterator
```

Comment:
still, if it is not super-ugly, I would refrain from implicits as they might confuse future maintainers of the test.